ios/Classes/ComWilliamrijksenOnesignalModule.m

```diff
@@ -13,93 +13,120 @@
 
-        if(payload.title) {
-            title = payload.title;
-        }
+    NSString* title = @"";
+    NSString* body = @"";
+    NSDictionary* additionalData = [[NSDictionary alloc] init];
 
-        if (payload.body) {
-            body = [payload.body copy];
-        }
+    if (payload.title) {
+        title = payload.title;
+    }
 
-        if (payload.additionalData) {
-            additionalData = payload.additionalData;
-        }
+    if (payload.body) {
+        body = [payload.body copy];
+    }
 
-        NSDictionary* notificationData = @{
-                                            @"title": title,
-                                            @"body": body,
-                                            @"additionalData": additionalData
-                                        };
-        [self fireEvent:@"notificationReceived" withObject:notificationData];
+    if (payload.additionalData) {
+        additionalData = payload.additionalData;
     }
-};
-    
-- (void) actionHandler:(OSNotificationOpenedResult *)result {
-    if ([self _hasListeners:@"notificationOpened"]) {
-        OSNotificationPayload* payload = result.notification.payload;
 
-        NSString* title = @"";
-        NSString* body = @"";
-        NSDictionary* additionalData = [[NSDictionary alloc] init];
+    NSDictionary* notificationData = @{
+                                        @"title": title,
+                                        @"body": body,
+                                        @"additionalData": additionalData
+                                    };
+    [self tryToPostNotification:notificationData withNotificationName:TiNotificationReceived];
+}
 
-        if(payload.title) {
-            title = payload.title;
-        }
++ (void) actionHandler:(OSNotificationOpenedResult *)result {
+    OSNotificationPayload* payload = result.notification.payload;
 
-        if (payload.body) {
-            body = [payload.body copy];
-        }
+    NSString* title = @"";
+    NSString* body = @"";
+    NSDictionary* additionalData = [[NSDictionary alloc] init];
 
-        if (payload.additionalData) {
-            additionalData = payload.additionalData;
-        }
+    if(payload.title) {
+        title = payload.title;
+    }
 
-        NSDictionary* notificationData = @{
-                                       @"title": title,
-                                       @"body": body,
-                                       @"additionalData": additionalData};
-        [self fireEvent:@"notificationOpened" withObject:notificationData];
+    if (payload.body) {
+        body = [payload.body copy];
     }
+
+    if (payload.additionalData) {
+        additionalData = payload.additionalData;
+    }
+
+    NSDictionary* notificationData = @{
+                                    @"title": title,
+                                    @"body": body,
+                                    @"additionalData": additionalData};
```

Comment:
@chmiiller This for example could be a dictionaty holding non-JS types like `NSData`, so ensure to map those (in this example, an `NSData` instance would be mapped with `[[TiBlob alloc] initWithData:data mimetype:@"text/plain"]`).